src/lattice.py

```diff
@@ -50,19 +59,28 @@ async def publish_entity(self, entity) -> Optional[Dict[str, Any]]:
         Raises:
             None
         """
+        session: Optional[aiohttp.ClientSession] = None
+        response: Optional[aiohttp.ClientResponse] = None
         try:
-            async with aiohttp.ClientSession(headers=self.generated_metadata) as session:
-                async with session.put(f"{self.lattice_ip}/api/v1/entities", json=entity) as response:
+            async with aiohttp.ClientSession(
+                headers=self.generated_metadata
+            ) as session:
+                async with session.put(
+                    f"{self.lattice_ip}/api/v1/entities", json=entity
+                ) as response:
                     response.raise_for_status()
                     entity_data = await response.json()
                     return entity_data
         except Exception as error:
             self.logger.error(
-                f"lattice api publish entity error {error}\n\trequest={session.request}\n\tresponse={response.status}\n\t{entity}")
+                f"lattice api publish entity error {error}\n\trequest={session.request if session else None}\n\tresponse={response.status if response else None}\n\t{entity}"
```

Comment:
The auto formatter obscures some of the changes here, but the changes are effectively these 2 lines, and the changes to line 62-63.